src/pages/SearchPage.tsx

```diff
@@ -21,7 +34,8 @@ const SearchPage = () => {
         `/search/all/${getSearchTerm}`
       );
       const postOnly = data.filter((el) => 'author' in el);
-      setPostResult(postOnly);
+      const posts = postOnly.map((post) => getPost(post._id));
+      Promise.all(posts).then((data) => setPostResult(data));
```

Comment:
`reduce`를 사용하게되면 1번의 탐색만으로도 원하는 데이터를 얻을 수 있으니 시간 복잡도가 줄어들 것 같습니다 😄 

```suggestion
      Promise.all(
        data.reduce<Promise<PostResponse>[]>((prev, post) => {
          if ('author' in post) {
            prev.push(getPost(post._id));
          }
          return prev;
        }, [])
      ).then((data) => setPostResult(data));
```

Reply:
오 reduce를 사용하셨군요 좋은 방법 같습니다! 반영해보겠습니다

Reply:
다만, 눈에 잘 읽히는 코드인지는 모르겠네요 😅